scalameta/parsers/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2941,7 +3000,7 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       if (isMutable) Defn.Var(mods, lhs, tp, rhs)
       else Defn.Val(mods, lhs, tp, rhs.get)
     } else {
-      mods.getAll[Mod.Lazy].foreach { m => syntaxError("lazy values may not be abstract", at = m) }
+      if (!isMutable) rejectMod[Mod.Lazy](mods, "lazy values may not be abstract")
```

Comment:
For correctness reasons, this block deals with both vars and vals. Lazy is only accepted for vals. So first I check that we're dealing with a val and then if `lazy` is present in the modifier lists, we reject it because `val` is just a declaration.